client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { RouteComponentProps } from 'react-router';
+import { Route, Redirect, RouteProps } from 'react-router-dom';
+import { useAuth } from '../../context/useAuthContext';
+
+interface ProtectedRoute extends RouteProps {
+  component: React.ComponentType<RouteComponentProps<any>> | React.ComponentType<any> | undefined;
+}
```

Comment:
`component` is already in the [`RouteProps` interface](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1ba317aa7d64467434e8f1fbb38bef2c6fb62e1d/types/react-router/index.d.ts#L87-L99) - it is optional there but see the comments below as to why we should keep it that way too!